packages/form-core/src/validate/LionValidationFeedback.js

```diff
@@ -1,37 +1,51 @@
 import { html, LitElement } from '@lion/core';
 
+/**
+ * @typedef {import('../validate/Validator').Validator} Validator
+ *
+ * @typedef {Object} messageMap
+ * @property {string} message
+ * @property {string} type
+ * @property {Validator} [validator]
+ */
+
 /**
  * @desc Takes care of accessible rendering of error messages
  * Should be used in conjunction with FormControl having ValidateMixin applied
  */
 export class LionValidationFeedback extends LitElement {
   static get properties() {
     return {
-      /**
-       * @property {FeedbackData} feedbackData
-       */
-      feedbackData: Array,
+      feedbackData: { attribute: false },
     };
   }
 
   /**
    * @overridable
+   * @param {Object} opts
+   * @param {string} opts.message
+   * @param {string} [opts.type]
+   * @param {Validator} [opts.validator]
    */
   // eslint-disable-next-line class-methods-use-this
   _messageTemplate({ message }) {
     return message;
   }
 
-  updated() {
-    super.updated();
+  /**
+   * @param {import('lit-element').PropertyValues } changedProperties
+   */
+  updated(changedProperties) {
+    super.updated(changedProperties);
     if (this.feedbackData && this.feedbackData[0]) {
       this.setAttribute('type', this.feedbackData[0].type);
       this.currentType = this.feedbackData[0].type;
       window.clearTimeout(this.removeMessage);
       if (this.currentType === 'success') {
         this.removeMessage = window.setTimeout(() => {
           this.removeAttribute('type');
-          this.feedbackData = '';
+          /** @type {messageMap[]} */
```

Comment:
Where is this type defined? Shouldn't custom types always be capitalized?

Reply:
Yeah as a convention we should probably capitalize. Defined at the top of the file, but perhaps better to put it in the types folder in one of the type definition files and import it always from this one place. I'm thinking ValidateMixinTypes.d.ts but I didn't create this file yet.